stdlib/2/os/path.pyi

```diff
@@ -22,25 +28,31 @@ devnull = ...  # type: str
 def abspath(path: AnyStr) -> AnyStr: ...
 def basename(path: AnyStr) -> AnyStr: ...
 
-def commonprefix(list: List[AnyStr]) -> AnyStr: ...
+if sys.version_info >= (3, 5):
+    def commonpath(paths: Sequence[AnyStr]) -> AnyStr: ...
+if sys.version_info >= (3,):
+    # NOTE: Empty List[bytes] results in '' (str) => fall back to Any return type.
+    def commonprefix(list: List[AnyStr]) -> Any: ...
+else:
+    def commonprefix(list: List[AnyStr]) -> AnyStr: ...
```

Comment:
Looking at the python2 version (because python3 does not let you mix types) it looks like the return type is the type of the `min` element of the list, https://github.com/python/cpython/blob/2.7/Lib/genericpath.py#L76

So maybe I can do an overload like this:

```
@overload
def commonprefix(list: List[AnyStr]) -> AnyStr: ...
@overload
def commonprefix(list: List[Union[bytes], Union[Text]]) -> Union[bytes, Text]
```

But then you have the dreaded `Union` in a return value.